tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -15049,5 +15120,400 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultiSpeedDXCoilsNoLoadFlowRateSiz
+     int CompIndex = 1;
+     bool HeatActive = false;
+     bool CoolActive = true;
+     int const ZoneOAUnitNum = 0;
+     Real64 const OAUCoilOutTemp = 0.0;
+     bool const ZoneEquipment = true;
+     Real64 sensOut = 0.0;
+     Real64 latOut = 0.0;
+
+     //Speed 1
+     thisSys->simulate(state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+     EXPECT_NEAR(thisSys->m_CycRatio, 0.02422, 0.001);
+     EXPECT_NEAR(sensOut, -227.705, 0.1);
+     DataGlobals::DoCoilDirectSolutions = true;
+     thisSys->FullOutput.resize(3);
+     thisSys->simulate(state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+     EXPECT_NEAR(thisSys->m_CycRatio, 0.02422, 0.001);
+     EXPECT_NEAR(sensOut, -227.705, 0.1);
+    // speed 2
+     DataZoneEnergyDemands::ZoneSysEnergyDemand(1).RemainingOutputRequired = -12000.0;
+     DataZoneEnergyDemands::ZoneSysEnergyDemand(1).RemainingOutputReqToCoolSP = -12000.0;
+     DataZoneEnergyDemands::ZoneSysEnergyDemand(1).RemainingOutputReqToHeatSP = -5000.0;
+     DataGlobals::DoCoilDirectSolutions = false;
+     thisSys->simulate(state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+     EXPECT_NEAR(thisSys->m_CycRatio, 1.000, 0.001);
+     EXPECT_NEAR(thisSys->m_SpeedRatio, 0.238771, 0.001);
+     EXPECT_NEAR(sensOut, -11998.0, 3.0);
+
+     DataGlobals::DoCoilDirectSolutions = true;
+     thisSys->simulate(state,
+                      thisSys->Name,
+                      FirstHVACIteration,
+                      AirLoopNum,
+                      CompIndex,
+                      HeatActive,
+                      CoolActive,
+                      ZoneOAUnitNum,
+                      OAUCoilOutTemp,
+                      ZoneEquipment,
+                      sensOut,
+                      latOut);
+     EXPECT_NEAR(thisSys->m_CycRatio, 1.000, 0.001);
+     EXPECT_NEAR(thisSys->m_SpeedRatio, 0.238771, 0.02);
+     EXPECT_NEAR(sensOut, -11998.0, 210.0);
```

Comment:
The Direct and Iterative solution methods results are identical. The unit test demonstrates the new method very well.

Reply:
:+1: